tedge/src/cli/disconnect/disconnect_bridge.rs

```diff
@@ -1,23 +1,42 @@
 use crate::cli::disconnect::error::*;
 use crate::command::*;
+use crate::services::tedge_mapper_az::TedgeMapperAzService;
 use crate::services::{
-    mosquitto::MosquittoService, tedge_mapper::TedgeMapperService, SystemdService,
+    mosquitto::MosquittoService, tedge_mapper_c8y::TedgeMapperC8yService, SystemdService,
 };
 use crate::utils::{paths, users::*};
 use which::which;
 
 const TEDGE_BRIDGE_CONF_DIR_PATH: &str = "mosquitto-conf";
 
+#[derive(Debug)]
+pub enum Cloud {
+    C8y,
+    Azure,
+}
+
+impl From<Cloud> for String {
```

Comment:
I am always having problems understanding code like `From<X> for Y` where the primary purpose is to convert X into Y (think of `toString()` methods in Java/Javacript etc.). 

Also implementing a trait for a remote type is something I am having slightly problems. This is now allowed in Rust, but it wasn't at all times. 

My personal preference is `impl Into<String> for Cloud`. But I understand that this will only allow you to call `.into()`, and not `String::from(cloud)`. 

Still, IMHO, from a readability pov, I highly prefer `Into<String> for Cloud`.